src/octets.rs

```diff
@@ -93,14 +94,16 @@ macro_rules! put_u {
             return Err(BufferTooShortError);
         }
 
+        let len = $len;
+
         #[allow(clippy::range_plus_one)]
-        let dst = &mut $b.buf[$b.off..($b.off + $len)];
+        let dst = &mut $b.buf[$b.off..($b.off + len)];
 
         unsafe {
             let src = &<$ty>::to_be($v) as *const $ty as *const u8;
```

Comment:
You've also got `$v` in this macro, sorry I missed that on first review.

Reply:
Fixed! thanks.